src/join.rs

```diff
@@ -0,0 +1,113 @@
+#[allow(unused)]
+use crate::Table;
+use crate::TableOption;
+use papergrid::{Entity, Grid};
+use std::cmp;
+
+pub struct Join {
+    pub mode: JoinMode,
+    pub strict_size: bool,
+}
+
+pub enum JoinMode {
+    Vertical(Table),
+    Horizontal(Table),
+}
+
+impl TableOption for Join {
+    fn change(&mut self, other: &mut Grid) {
+        match &mut self.mode {
+            JoinMode::Vertical(table) if self.strict_size => {
+                let new_row_size = table.grid.count_rows() + other.count_rows();
+                let mut new_grid = Grid::new(new_row_size, table.grid.count_rows());
```

Comment:
It seems to me this is the only line which is different from the other `Vertical` branch?
It may be worth to combine these 2 to eliminate duplication?
I might be wrong though as I haven't thoroughly checked it.

https://github.com/zhiburt/tabled/blob/e66253b35bc17de83a80ec8f39e82c7c8c82064d/src/join.rs#L64

Reply:
Since other.count_rows()is equal to cmp::max(other.count_rows(),table.grid.count_column()) here, I used other.count_rows() instead of new_row_size.
This may cause a misunderstanding like this time, so I'll fix it.